include/grpc/event_engine/event_engine.h

```diff
@@ -255,6 +255,24 @@ class EventEngine : public std::enable_shared_from_this<EventEngine> {
     /// values are expected to remain valid for the life of the Endpoint.
     virtual const ResolvedAddress& GetPeerAddress() const = 0;
     virtual const ResolvedAddress& GetLocalAddress() const = 0;
+
+   protected:
+    /// An internal method which takes an extension name as input and returns a
+    /// valid pointer if the extension is supported by the Endpoint
+    /// implementation.
+    virtual void* BasicQueryExtension(absl::string_view id) { return nullptr; }
+
+   public:
+    /// A method which allows users to query whether an Endpoint implementation
+    /// supports a specified extension.
+    ///
+    /// The type of the extension is encoded in the template parameter and
+    /// Endpoint implementations which support the desired extension can
+    /// override the BasicQueryExtension method to return a valid pointer.
+    template <typename T>
```

Comment:
If this is template is going to remain here, the type should be explained with an example in the documentation.

Reply:
Per our offline discussion, moved the template as a helper function to src/core/lib/event_engine/query_extensions.h